src/Image.zig

```diff
@@ -42,17 +44,49 @@ pub const Stream = io.StreamSource;
 
 pub const EncoderOptions = AllImageFormats.ImageEncoderOptions;
 
-pub const SaveInfo = struct {
-    width: usize,
-    height: usize,
-    encoder_options: EncoderOptions,
+pub const AnimationLoopInfinite = -1;
+
+pub const AnimationFrame = struct {
+    pixels: PixelStorage,
+    duration: f32,
+};
+
+pub const Animation = struct {
+    frames: std.ArrayListUnmanaged(AnimationFrame),
+    loop_count: i32 = AnimationLoopInfinite,
+
+    pub fn deinit(self: *Animation, allocator: std.mem.Allocator) void {
+        for (self.frames.items) |frame| {
+            frame.pixels.deinit(allocator);
+        }
+
+        self.frames.deinit(allocator);
+    }
+};
+
+pub const Data = union(enum) {
+    empty: void,
+    image: PixelStorage,
+    animation: Animation,
+
+    pub fn deinit(self: *Data, allocator: std.mem.Allocator) void {
```

Comment:
We should probably provide init methods here and above as well, not just for symmetry but for generating new animated images that we then want to serialize.

Reply:
Will do